public/app/plugins/datasource/grafana-azure-monitor-datasource/composable_dataquery.cue

```diff
@@ -0,0 +1,188 @@
+							resourceUri?: string
+
+							// @deprecated Use resources instead 
+							resourceGroup?: string
+							// @deprecated Use resources instead 
+							resourceName?: string
+						} @cuetsy(kind="interface")
+
+						// Azure Monitor Logs sub-query properties
+						#AzureLogsQuery: {
+							query?:        string
+							resultFormat?: string
+							resources?: [...string]
+
+							workspace?: string
+
+							// @deprecated Use resources instead 
+							resource?: string
+						} @cuetsy(kind="interface")
+
+						#AzureResourceGraphQuery: {
+							query?:        string
+							resultFormat?: string
+						} @cuetsy(kind="interface")
+
+						#AzureMonitorResource: {
+							subscription?:    string
+							resourceGroup?:   string
+							resourceName?:    string
+							metricNamespace?: string
+							region?:          string
+						} @cuetsy(kind="interface")
+
+						#AzureMetricDimension: {
+							dimension: string
+							operator:  string
+							filters?: [...string]
+							// @deprecated filter is deprecated in favour of filters to support multiselect
+							filter?: string
+						} @cuetsy(kind="interface")
+
+						#GrafanaTemplateVariableQueryType: "AppInsightsMetricNameQuery" | "AppInsightsGroupByQuery" | "SubscriptionsQuery" | "ResourceGroupsQuery" | "ResourceNamesQuery" | "MetricNamespaceQuery" | "MetricNamesQuery" | "WorkspacesQuery" | "UnknownQuery" @cuetsy(kind="type")
+						#BaseGrafanaTemplateVariableQuery: {
+							rawQuery?: string
+							...
+						} @cuetsy(kind="interface")
+						#UnknownQuery: #BaseGrafanaTemplateVariableQuery & {
+										kind: "UnknownQuery"
+						}                            @cuetsy(kind="interface")
+						#AppInsightsMetricNameQuery: #BaseGrafanaTemplateVariableQuery & {
+										kind: "AppInsightsMetricNameQuery"
+						}                         @cuetsy(kind="interface")
+						#AppInsightsGroupByQuery: #BaseGrafanaTemplateVariableQuery & {
+									kind:       "AppInsightsGroupByQuery"
+									metricName: string
+						}                    @cuetsy(kind="interface")
+						#SubscriptionsQuery: #BaseGrafanaTemplateVariableQuery & {
+									kind: "SubscriptionsQuery"
+						}                     @cuetsy(kind="interface")
+						#ResourceGroupsQuery: #BaseGrafanaTemplateVariableQuery & {
+									kind:         "ResourceGroupsQuery"
+									subscription: string
+						}                    @cuetsy(kind="interface")
+						#ResourceNamesQuery: #BaseGrafanaTemplateVariableQuery & {
+									kind:            "ResourceNamesQuery"
+									subscription:    string
+									resourceGroup:   string
+									metricNamespace: string
+						}                      @cuetsy(kind="interface")
+						#MetricNamespaceQuery: #BaseGrafanaTemplateVariableQuery & {
+							kind:             "MetricNamespaceQuery"
+							subscription:     string
+							resourceGroup:    string
+							metricNamespace?: string
+							resourceName?:    string
+						} @cuetsy(kind="interface")
```

Comment:
I believe all of these should be deprecated? (it's okay to leave them if you are just copying what we have in TS code).

Reply:
Yep they're deprecated, I'll validate that they're unused and remove them (and any associated logic).

Reply:
If you're able to just pull these out now, awesome - i'm glad that schematization could be a forcing function to clean up some code :)

But know that as we turn on use of multi-version schemas via thema in the coming months, there will be new strategies available for handling deprecation of query types. So if trying to remove these now causes headaches when thinking about e.g. backwards compatibility, i'd say it's fine to hold off because the set of tools you have available is about to change